src/main/java/spiffe/api/svid/WorkloadAPIClient.java

```diff
@@ -0,0 +1,96 @@
+package spiffe.api.svid;
+
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spiffe.api.svid.util.ExponentialBackOff;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static spiffe.api.svid.Workload.*;
+
+/**
+ * Provides functionality to interact with a Workload API
+ *
+ */
+public final class WorkloadAPIClient {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(WorkloadAPIClient.class);
+
+    private SpiffeWorkloadStub spiffeWorkloadStub;
+
+    /**
+     * Constructor
+     * @param spiffeEndpointAddress
+     */
+    public WorkloadAPIClient(String spiffeEndpointAddress) {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(spiffeEndpointAddress);
+    }
+
+    /**
+     * Default constructor
+     * The WorkloadAPI Address will be resolved by an Environment Variable
+     *
+     */
+    public WorkloadAPIClient() {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(EMPTY);
+    }
+
+    /**
+     * Fetch the SVIDs from the Workload API on a synchronous fashion
+     * Use a Exponential Backoff to handle the errors and retries
+     *
+     * @return List of X509SVID or Empty List if none have been fetched
+     */
+    public List<X509SVID> fetchX509SVIDs() {
+        try {
+            return ExponentialBackOff.execute(this::callWorkloadStub_fetchX509SVIDs);
+        } catch (Exception e) {
+            LOGGER.error("Couldn't get SVIDs from Workload API", e);
+            return Collections.emptyList();
+        }
+    }
+
+    /**
+     * Fetch the SVIDs from the Workload API on a asynchronous fashion
+     *
+     * TODO: Use a Exponential Backoff to handle the errors and retries
```

Comment:
This feels pretty important - I would not call this lib production ready w/o it. We want workloads to be able to survive and gracefully recover from transient conditions like agent restarts etc.

Reply:
yes, definitely, errors handling is very important, I'm working on how to make the retries on the asynchronous calls

Reply:
What about using failure handling library like https://github.com/jhalterman/failsafe ? The java client could define a default RetryPolicy and users could also define their own and pass it to the client.

Reply:
I added a retry on the onError() of the StreamObserver to basically retry on every error that may happen except when it's a InvalidArgument. 
@eicnix, would you want to suggest how to improve it using a handling library? It would be very helpful.